### PR TITLE
solved the image pull auth issue by adding oauthscopes in api request

### PR DESCRIPTION
**Pull request by leiyiz**
/kind bug

**What this PR does / why we need it**:
in poc work previously, the image cannot be pulled because of some auth issue, this pr fixed that issue

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

-->
```release-note
NONE
```

### CODE REVIEW

File: test/k8s-integration/cluster.go [hunk 1]
```diff
@@ -307,6 +307,12 @@ func clusterUpGKE(gceZone, gceRegion string, numNodes int, imageType string, use
 				NodeConfig: &container.NodeConfig{
 					MachineType: "n1-standard-2",
 					ImageType:   imageType,
+					OauthScopes: []string{
+						"https://www.googleapis.com/auth/devstorage.read_only",
```
[the remark this answers is not in the document]
Comment (saikat-royc) [reply test/k8s-integration/cluster.go:311]: I think only www.googleapis.com/auth/devstorage.read_only is the important one.
https://github.com/googleapis/google-api-go-client/blob/master/container/v1beta1/container-gen.go#L3406 leiyi pointed to this.

We can also verify this locally by deploying a cluster without filestorecsi, and make sure pdcsi pods comeup.

[the remark this answers is not in the document]
Comment (leiyiz) [reply test/k8s-integration/cluster.go:311]: I looked at when I use `gcloud` to create a gke cluster, these are the scopes added by gcloud: https://diff.googleplex.com/#key=cfcFhAhdYJk0&diffId=13. I think it'd be good to see if gcloud has logic to add different scope for different cluster creations. odd that gcloud did not add `auth/compute` which is apparently required for mounting persistent storage onto node

[the remark this answers is not in the document]
Comment (leiyiz) [reply test/k8s-integration/cluster.go:311]: and yes with these scope all pods does come up

File: test/k8s-integration/cluster.go [hunk 2]
```diff
@@ -307,6 +307,9 @@ func clusterUpGKE(gceZone, gceRegion string, numNodes int, imageType string, use
...
```
Comment (mattcary) [inline test/k8s-integration/cluster.go:311]: I just remembered that https://www.googleapis.com/auth/cloud-platform is the uber-scope that everyone seems to be moving to these days.

There more discussion about this here: https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances#using

> Generally, you can just set the cloud-platform access scope to allow access to most of the Cloud APIs, then grant the service account only relevant IAM roles. The combination of access scopes granted to the virtual machine instance and the IAM roles granted to the service account determines the amount of access the service account has for that instance. The service account can execute API methods only if they are allowed by both the access scope and its IAM roles.

**Pull request merged**